### PR TITLE
Allow delete to abort an in-progress spawn

### PR DESCRIPTION
**Pull request by rra**
Previously, we returned 409 if someone attempted to delete a spawn that was still in progress. This interferes with cleanup that we want to do in the REST Spawner if JupyterHub was restarted while spawns are in progress. JupyterHub may lose the credentials for those labs and need to stop them so that they can be restarted with correct credentials, but the previous constraint meant that the stop would fail until the spawn completed.

Use the new cancel method on the monitor to stop any spawn in progress and then continue with a delete to clean up any remnant Kubernetes resources. Send a failure event to the event stream of any in-progress operation when the operation is cancelled.

### CODE REVIEW

File: tests/handlers/labs_test.py [hunk 1]
```diff
@@ -560,7 +586,7 @@ async def test_spawn_errors(
     mock_slack: MockSlackWebhook,
 ) -> None:
     lab = read_input_lab_specification_json("base", "lab-specification.json")
-    apis_to_fail = {"read_namespaced_secret"}
+    apis_to_fail = set()
```
Comment (athornton) [inline tests/handlers/labs_test.py:589]: Is this change unrelated?  If not, I don't understand what reading a namespaced secret has to do with it.

File: src/jupyterlabcontroller/services/lab.py [hunk 2]
```diff
@@ -1015,16 +1028,28 @@ def in_progress(self) -> _LabOperation | None:
     async def cancel(self) -> None:
         """Cancel any existing operation.
 
-        This is called during process shutdown. All operations in progress are
-        cancelled, which may strand random lab state that we will detect and
-        clean up during reconciliation.
+        This is called during process shutdown and to abort a spawn in
```
Comment (athornton) [inline src/jupyterlabcontroller/services/lab.py:1031]: Wait a minute.  I may have just realized why it's mostly the bots that are exploding.

The JupyterHub UI doesn't let you delete a spawn that's in progress...but we might bypass that with mobu.

**Review by athornton — APPROVED** [review]
LGTM, and maybe I understood something new.

File: tests/handlers/labs_test.py (hunk 1 above)
Comment (rra) [reply tests/handlers/labs_test.py:589]: Sorry, that was me being lazy. Unrelated (but correct) change left over from testing something else.

File: src/jupyterlabcontroller/services/lab.py (hunk 2 above)
Comment (rra) [reply src/jupyterlabcontroller/services/lab.py:1031]: mobu only interacts with JupyterHub, so if JupyterHub doesn't permit it, mobu can't do it.

**Pull request merged**